tests/jni_api.rs

```diff
@@ -600,6 +592,27 @@ pub fn throw_new_fail() {
     assert_pending_java_exception(&env);
 }
 
+#[test]
+pub fn throw_defaults() {
+    let env = attach_current_thread();
+
+    assert_throws(&env, "Default exception thrown");
+    assert_throws(&env, "Default exception thrown".to_owned());
+    assert_throws(&env, JNIString::from("Default exception thrown"));
+}
+
+fn assert_throws<'a, D>(env: &JNIEnv<'a>, descriptor: D)
```

Comment:
This method does not only assert, it also performs the test itself, hence the name is slightly misleading. Also, it does not accept the expected values, but has them as locals, which is implicit and duplicates what is in `throw_defaults`.

I suggest making the expected value explicit, and passing the thing to assert on directly (or renaming this method to `test_throwable_descriptors_with_default_type(env, descriptor, expected_message)`). Or use constants.

Reply:
It looks better with constants.